### PR TITLE
[Merged by Bors] - chore: rename `elementalStarAlgebra` and parallelize API

### PR DESCRIPTION
**Pull request by j-loreaux**
#18615 adds non-unital versions of the `elemental` (star) algebras: i.e., the topological closure of the algebra generated by a single element.

That PR develops a series of standard lemmas and uniform naming practices. This PR adapts the existing unital `elemental` algebras to the naming convention and adjusts the existing lemmas to match API.

---

Note: this is independent of #18615

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Algebra/Algebra.lean [hunk 1]
```diff
@@ -574,6 +574,53 @@ theorem Subalgebra.topologicalClosure_comap_homeomorph (s : Subalgebra R A) {B :
...
+@[aesop safe apply (rule_sets := [SetLike])]
+theorem self_mem (x : A) : x ∈ elemental R x :=
+  le_topologicalClosure _ <| self_mem_adjoin_singleton R x
+
+@[deprecated (since := "2024-11-05")] alias _root_.Algebra.self_mem_elementalAlgebra := self_mem
+
+theorem le_of_mem {x : A} {s : Subalgebra R A} (hx : x ∈ s)
+    (hs : IsClosed (s : Set A)) :
+    elemental R x ≤ s :=
+  topologicalClosure_minimal _ (adjoin_le <| by simpa using hx) hs
```
Comment (ADedecker) [inline Mathlib/Topology/Algebra/Algebra.lean:598]: Could you also add the `iff` statement ? These tend to be very useful when rewriting.

File: Mathlib/Topology/Algebra/Algebra.lean [hunk 2]
```diff
@@ -574,6 +574,53 @@ theorem Subalgebra.topologicalClosure_comap_homeomorph (s : Subalgebra R A) {B :
...
+instance isClosed (x : A) : IsClosed (elemental R x : Set A) :=
+  isClosed_topologicalClosure _
+
+instance [T2Space A] {x : A} : CommSemiring (elemental R x) :=
+  commSemiringTopologicalClosure _
+    letI : CommSemiring (adjoin R {x}) :=
+      adjoinCommSemiringOfComm R fun y hy z hz => by
+        rw [mem_singleton_iff] at hy hz
+        rw [hy, hz]
+    fun _ _ => mul_comm _ _
```
Comment (ADedecker) [inline Mathlib/Topology/Algebra/Algebra.lean:609]: I'm a bit scared by the fact that [Subalgebra.commSemiringTopologicalClosure](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Algebra.html#Subalgebra.commSemiringTopologicalClosure) is not reducible (see e.g [Subalgebra.commRingTopologicalClosure](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Algebra.html#Subalgebra.commRingTopologicalClosure) for comparison). Would it make sense to explicitly override the `toSemiring` field to make sure all data is easily reducible to what it should be?

Comment (j-loreaux) [reply Mathlib/Topology/Algebra/Algebra.lean:609]: oh, I think those should just be marked `abbrev` instead. I'll make that change later.

Comment (j-loreaux) [reply Mathlib/Topology/Algebra/Algebra.lean:609]: I've done this in #18700 (along with other things)

File: Mathlib/Topology/Algebra/StarSubalgebra.lean [hunk 3]
```diff
@@ -170,71 +170,87 @@ end StarSubalgebra
...
-theorem star_self_mem (x : A) : star x ∈ elementalStarAlgebra R x :=
+theorem star_self_mem (x : A) : star x ∈ elemental R x :=
   star_mem <| self_mem R x
 
-/-- The `elementalStarAlgebra` generated by a normal element is commutative. -/
-instance [T2Space A] {x : A} [IsStarNormal x] : CommSemiring (elementalStarAlgebra R x) :=
+@[deprecated (since := "2024-11-05")]
+alias _root_.elementalStarAlgebra.star_self_mem := star_self_mem
+
+/-- The `elemental` generated by a normal element is commutative. -/
```
Comment (ADedecker) [inline Mathlib/Topology/Algebra/StarSubalgebra.lean:202]:
```suggestion
/-- The `elemental` star subalgebra generated by a normal element is commutative. -/
```

**Review by ADedecker — APPROVED** [review]
Thanks!

bors d+

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**